server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -297,6 +298,81 @@ void executeOnShardTarget(
         );
     }
 
+    public void executeRequest(
```

Comment:
This part of code was used for create pits api. But since the pit changes are feature branch, somebody removed this code citing its not used anywhere, so readded it back

Reply:
I don't understand, if this code was removed, it should have resulted in compile error.
Anyway, this can be added back as a separate PR. This PR itself is a big PR to review.